core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -480,6 +480,11 @@ public TableScan newScan() {
       throw new UnsupportedOperationException("Transaction tables do not support scans");
     }
 
+    @Override
+    public TableScan appendsBetween(long fromSnapshotId, long toSnapshotId) {
+      throw new UnsupportedOperationException("Transaction tables do not support incremental scans");
+    }
+
```

Comment:
Should we explicitly block `appendsAfter` as well? Just in case the default implementation on the interface changes.

Reply:
+1. I'll make that change

Reply:
Fixed this under `BaseTableScan` as these apis have move under `TableScan`